ftw/publisher/core/adapters/dx_field_data.py

```diff
@@ -151,8 +166,21 @@ def unpack(self, name, field, value):
                 schema.interfaces.ITime,
                 schema.interfaces.IDatetime,
                 ]):
-            if value:
-                return DateTime.DateTime(value).asdatetime()
+            if value and value.get('date_or_time'):
+                # Expect ISO-8601 datetime/time/date where zone was stripped (not recalculated)
+                date_or_time = dateutil.parser.parse(value['date_or_time'])
+                zone_str = value.get('zone')
+
+                if self._provided_by_one_of(field, [schema.interfaces.IDate]):
+                    return date_or_time.date()
+
+                if zone_str:
+                    date_or_time = pytz.timezone(zone_str).localize(date_or_time)
+
+                if self._provided_by_one_of(field, [schema.interfaces.ITime]):
```

Comment:
providedBy(...) would work just as well here (and line 174)

Reply:
Done